packages/core/lib/test.js

```diff
@@ -98,26 +98,31 @@ const Test = {
     );
     testResolver.cache_on = false;
 
-    const { compilations } = await this.compileContractsWithTestFilesIfNeeded(
-      solTests,
-      config,
-      testResolver
-    );
-
-    const testContracts = solTests.map(testFilePath => {
-      return testResolver.require(testFilePath);
-    });
-
     const runner = new TestRunner(config);
+    var compilations = null;
+
+    // Do not compile contracts, test contracts, do not add test contracts to testing set
+    // when --compile-none flag is set
+    if (!config.compileNone) {
+      var { compilations } = await this.compileContractsWithTestFilesIfNeeded(
```

Comment:
Instead of declaring `compilations` twice, could you move this `compileNone` check to inside the `compileContractsWithTestFilesIfNeeded` method?

Reply:
Ok. I can return {} if config.compileNone is set to true. Does it make sense ? 
I will still have to make check for config.compileNone flag in Test.run function to skip solidity tests if this flag is set.